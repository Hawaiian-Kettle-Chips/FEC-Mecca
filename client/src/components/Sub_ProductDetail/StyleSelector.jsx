import React from 'react';
import StyleList from './Sub_StyleSelector/StyleList.jsx';

function StyleSelector({
  productStyle, currentStyle, setCurrentStyle, startSelect, setStartSelect,
}) {
  const { sale_price, original_price, name } = currentStyle;
  return (
    <div className="style-selector">
      <div className="price" style={{ padding: '1rem 0' }}>
        {sale_price
          ? (
            <p>
              <span className="sale-price">${sale_price}</span>
              <span className="original-sale-price">${original_price}</span>
            </p>
          )
          : (
            <p>
              <span className="original-price">${original_price}</span>
            </p>
          )}
      </div>
      <div style={{ paddingBottom: '1rem' }}>
        STYLE {'>'} &nbsp;
        <span style={{ fontStyle: 'italic' }}>{startSelect ? name : 'SELECTED STYLE'}</span>
      </div>
      <StyleList
        productStyle={productStyle}
        setCurrentStyle={setCurrentStyle}
        currentStyle={currentStyle}
        startSelect={startSelect}
        setStartSelect={setStartSelect}
      />
    </div>
  );
}
export default StyleSelector;
