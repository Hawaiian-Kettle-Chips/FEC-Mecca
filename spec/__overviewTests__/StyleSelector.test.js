import React from 'react';
import { cleanup, render, screen, within, act } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { beforeEach, afterEach, describe, it, expect } from '@jest/globals';
import axios from 'axios';
import 'regenerator-runtime/runtime';
import StyleSelector from '../../client/src/components/Sub_ProductDetail/StyleSelector';

require('dotenv').config();

afterEach(cleanup);
describe('test style selector', () => {
  let container;
  let productStyle;
  let currentStyle;
  const startSelect = false;
  const setCurrentStyle = jest.fn();
  const setStartSelect = jest.fn();

  beforeEach(async () => {
    const baseURL = 'https://app-hrsei-api.herokuapp.com/api/fec2/hr-rfp';
    const productID = 40349;
    const url = `/products/${productID}/styles`;
    const options = {
      method: 'get',
      baseURL,
      url,
      headers: {
        Authorization: process.env.API_KEY,
      },
    };
    const response = await axios(options);
    productStyle = response.data;
    const styles = productStyle.results;

    const defaultStyle = styles.filter((style) => style['default?']);
    currentStyle = defaultStyle[0];
    const { container: renderedContainer } = render(<StyleSelector
      productStyle={productStyle}
      currentStyle={currentStyle}
      setCurrentStyle={setCurrentStyle}
      startSelect={startSelect}
      setStartSelect={setStartSelect}
    />);
    container = renderedContainer;
  });

  it('Render the style selector', async () => {
    expect(container).toMatchSnapshot();
  });

  it('Render the style selector', async () => {
    expect(container).toMatchSnapshot();
  });


});
