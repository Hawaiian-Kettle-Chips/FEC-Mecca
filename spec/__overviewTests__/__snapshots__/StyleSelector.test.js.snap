// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test style selector Render the style selector 1`] = `
<div>
  <div
    class="style-selector"
  >
    <div
      class="price"
      style="padding: 1rem 0px;"
    >
      <p>
        <span
          class="original-price"
        >
          $
          89.00
        </span>
      </p>
    </div>
    <div
      style="padding-bottom: 1rem;"
    >
      STYLE 
      &gt;
       Â 
      <span
        style="font-style: italic;"
      >
        SELECTED STYLE
      </span>
    </div>
    <div
      class="style-list-container"
    >
      <div
        class="style-entry-container"
        style="background-image: url(https://images.unsplash.com/photo-1477420143023-6a0e0b04b69a?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80);"
      />
      <div
        class="style-entry-container"
        style="background-image: url(https://images.unsplash.com/photo-1463100099107-aa0980c362e6?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80);"
      />
    </div>
  </div>
</div>
`;
